waspc/src/Wasp/Generator/WebAppGenerator/Common.hs

```diff
@@ -89,8 +96,20 @@ mkTmplFdWithDstAndData ::
   Path' (Rel WebAppRootDir) File' ->
   Maybe Aeson.Value ->
   FileDraft
-mkTmplFdWithDstAndData srcPathInWebAppTemplatesDir dstPathInWebAppRootDir tmplData =
+mkTmplFdWithDstAndData = mkAnyValidTmplFdWithDstAndData webAppTemplatesDirInTemplatesDir
+
+mkSharedTmplFdWithDst :: Path' (Rel SharedTemplatesDir) File' -> Path' (Rel WebAppRootDir) File' -> FileDraft
+mkSharedTmplFdWithDst relSrcPath relDstPath = mkAnyValidTmplFdWithDstAndData sharedTemplatesDirInTemplatesDir relSrcPath relDstPath Nothing
+
+mkAnyValidTmplFdWithDstAndData ::
+  ValidWebAppTemplatesDir d =>
+  Path' (Rel TemplatesDir) (Dir d) ->
+  Path' (Rel d) File' ->
+  Path' (Rel WebAppRootDir) File' ->
+  Maybe Aeson.Value ->
+  FileDraft
+mkAnyValidTmplFdWithDstAndData templatesDir relSrcPath relDstPath tmplData =
```

Comment:
I think the naming is a bit off here.

`mkAnyValidTmplFdWithDstAndData` (has prefix `AnyValid`) specializes into:
  - `mkSharedTmplFdWithDst` (has prefix `Shared`)
  - `mkTmplFdWithDstAndData` (no prefix, implying it's the general one of the three, but is actually specific).

I suggest:

`mkTmplFdWithDstAndData`specializes into:
  - `mkSharedTmplFdWithDstAndData`
  - `mkWebAppTmplFdWithDstAndData`

Reply:
I wanted to avoid renaming the most used function to `mkWebAppTmplFdWithDstAndData`, I liked your idea about ditching polymorphism and just created a concrete function related to using the shared templates. 

I'll see how the renaming goes and take this comment into account 👍 "more specialized function shouldn't look like the more general function due to a missing prefix"

Reply:
There are now two independnt functions `mkTmplFdWithDst` like before (which lives in either `Server` or `WebApp` context) and a special one `mkSharedTmplFdWithDst` which lives alongside. the frist one, but it signals that it uses the shared templates vs. regular templates.